inst/stan/wwinference.stan

```diff
@@ -187,9 +190,9 @@ transformed parameters {
   s = get_vl_trajectory(t_peak, viral_peak, dur_shed, gt_max);
 
   // Site level disease dynamic estimates!
-  i0_site_over_n = inv_logit(logit(i0_over_n) + eta_i0 * sigma_i0);
-  growth_site = initial_growth + eta_growth * sigma_growth; // site level growth rate
   for (i in 1:n_subpops) {
+    vector[n_weeks] log_r_site_t_in_weeks;
```

Comment:
Again, slight preference for all variables, even ones that are getting dynamically updated like this one, to be defined in the beginning of the block

Reply:
FYI this has an impact on memory as anything stored outside of brackets (even just empty brackets) is stored and anything inside isn't. This can also have a slight performance impact.

Reply:
Related: I think defining throwaway loop variables within the loop makes code slightly clearer. It indicates to the reader that those variables are throwaways and won't be reused/stored.

Reply:
@seabbs I'm not sure I understand your comment. Do you mean that this isn't stored because it's inside brackets (in this case a for loop?). 

I am fine with that @dylanhmorris I think that makes sense

Reply:
> Do you mean that this isn't stored because it's inside brackets (in this case a for loop?).

Yes it makes it a temporary variable that being said I wouldn't redefine it inside a loop like this as it will an a small allocation overhead each time. Instead if I felt worried I would do { int temp; for (.... temp = 1, use temp)}

Reply:
The compiler reallocates? That's surprising to me. I'd have expected it to work out that it could allocate only on the first iteration and then overwrite.